src/udiskslinuxblock.c

```diff
@@ -3248,6 +3248,48 @@ handle_format (UDisksBlock           *block,
 
 /* ---------------------------------------------------------------------------------------------------- */
 
+static gint
+open_device (const gchar      *device,
+             const gchar      *mode,
+             gint              flags,
+             GError          **error)
+{
+  gint fd = -1;
+
+  if (flags & O_RDWR || flags & O_RDONLY || flags & O_WRONLY)
+    {
+      g_set_error (error, UDISKS_ERROR, UDISKS_ERROR_FAILED,
+                   "Using 'O_RDWR', 'O_RDONLY' and 'O_WRONLY' flags is not permitted.\
+                    Use 'mode' argument instead.");
+      goto out;
```

Comment:
How is this better than `return fd;`? Yeah I see this seems to be convention in this code-base.